src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -167,6 +170,11 @@ def __post_init__(self):
                 f"Given platform constraints {self.platforms} for internal only pex request: "
                 f"{self}."
             )
+        if self.requirements and self.requirements_file:
+            raise ValueError(
+                "You should only specify `requirements` or `requirements_file`, but both were set: "
+                f"{self}"
+            )
```

Comment:
From a modeling perspective, is this a useful case for a `union`?

Reply:
In a followup, I move this setting from `PexRequest` into `PexRequirements`, which allows you to set `req_strings`, `file_content`, or `file_path`. The modeling works quite well there.